packages/react-scripts/config/modules.js

```diff
@@ -71,7 +72,10 @@ function getModules() {
   // TypeScript project and set up the config
   // based on tsconfig.json
   if (hasTsConfig) {
-    config = require(paths.appTsConfig);
+    const ts = require(resolve.sync('typescript', {
```

Comment:
Why are you using `resolve.sync` here?

Reply:
I saw this approach being used in [`verifyTypeScriptSetup`](https://github.com/facebook/create-react-app/blob/master/packages/react-scripts/scripts/utils/verifyTypeScriptSetup.js#L62-L64) and thought  that there must be a reason `require` isn't used directly.

I'm not 100% certain but it seems like it might have to do with how [`appDirectory`](https://github.com/facebook/create-react-app/blob/master/packages/react-scripts/config/paths.js#L15-L17) is setup to resolve symlinks?

Reply:
I be leave this is necessary to support yarn pnp. As typescript, isn't a dependency of CRA, yarn would not let CRA require it directly. However, by resolving it from the app directory (which do have a dependency on typescript), CRA can still access it.